File: back-end/PHP/yii/backend/controllers/ChatController.php
<?php
/**
 * Created by huxin.
 * User: huxin
 * Date: 2019-02-27
 * Time: 09:41
 */

namespace backend\controllers;


use app\models\Chat;
use backend\exception\ValidateException;

class ChatController extends BasicController
{
    public $enableCsrfValidation = false;
    private $chat;

    public function init()
    {
        parent::init();
        $this->chat = new Chat();
    }

    /**
     * 获取说说列表
     * Date: 2019-02-27 10:09
     * @return \yii\web\Response
     * @throws ValidateException
     */
    public function actionChatList()
    {
        $this->basicValidate($this->chat, 'chatList');
        $list = $this->chat->find()
            ->offset((($this->request_data)['pageNum'] - 1) * ($this->request_data)['pageSize'])
            ->limit(($this->request_data)['pageSize'])
            ->orderBy('created_at DESC')
            ->all();
        return !empty($list) ? $this->success('获取说说列表成功！', $list)
            : $this->error('暂无说说列表数据！');
    }

    /**
     * 添加说说
     * Date: 2019-02-27 10:09
     * @return \yii\web\Response
     * @throws ValidateException
     */
    public function actionChatAdd()
    {
        $chat = $this->basicValidate($this->chat, 'addChat');
        $res = $chat->save(false);
        return $res ? $this->success('添加标签成功！')
            : $this->error('添加标签失败，请稍后再试！');
    }

    /**
     * 更新说说
     * Date: 2019-02-27 10:09
     * @return \yii\web\Response
     * @throws ValidateException
     */
    public function actionChatUpdate()
    {
        $this->basicValidate($this->chat, 'updateChat');
        $exist_chat = $this->chat->findOne(($this->request_data)['id']);
        $exist_chat->content = ($this->request_data)['content'];
        $exist_chat->is_show = ($this->request_data)['is_show'];
        $res = $exist_chat->save(false, ['content', 'is_show', 'updated_at']);
        return $res ? $this->success('修改标签成功！')
            : $this->error('修改标签失败，请稍后再试！');
    }

    /**
     * 删除说说
     * Date: 2019-02-27 10:10
     * @return \yii\web\Response
     * @throws ValidateException
     */
    public function actionDelChat()
    {
        $this->basicValidate($this->chat, 'chatDelete');
        $exist_chat = $this->chat->findOne(($this->request_data)['id']);
        $res = $exist_chat->delete();
        return $res ? $this->success('删除标签成功！')
            : $this->error('删除标签失败，请稍后再试！');
    }
}

File: back-end/Go/iris/controllers/backend/chatController.go
package backend

import (
	"blog/controllers"
	"blog/models"
	"github.com/kataras/iris/v12"
)

/**
获取说说列表
*/
func GetChatList(ctx iris.Context) {
	pageNum := ctx.URLParamInt64Default("pageNum", 1)
	pageSize := ctx.URLParamInt64Default("pageSize", 10)
	list := models.GetChatList(pageNum, pageSize)
	if len(list) > 0 {
		response.RenderSuccess(ctx, "获取说说列表成功", list)
		return
	}
	response.RenderError(ctx, "暂无说说列表数据", nil)
}

/**
添加说说
*/
func AddChat(ctx iris.Context) {
	chat := models.Chat{}
	fields := []string{"is_show", "content"}
	validateFields := []string{"IsShow", "Content"}
	_, err := controllers.GetRightModel(ctx, &chat, fields, validateFields)
	if err != nil {
		response.RenderError(ctx, err.Error(), nil)
		return
	}
	res := models.AddChat(&chat)
	if !res {
		response.RenderError(ctx, "添加说说失败，请稍后再试！", nil)
		return
	}
	response.RenderSuccess(ctx, "添加说说成功！", nil)
}

/**
修改说说
*/
func UpdateChat(ctx iris.Context) {
	chat := models.Chat{}
	fields := []string{"id", "is_show", "content"}
	validateFields := []string{"ID", "IsShow", "Content"}
	_, err := controllers.GetRightModel(ctx, &chat, fields, validateFields)
	if err != nil {
		response.RenderError(ctx, err.Error(), nil)
		return
	}
	res := models.UpdateChat(&chat)
	if !res {
		response.RenderError(ctx, "修改说说失败，请稍后再试！", nil)
		return
	}
	response.RenderSuccess(ctx, "修改说说成功！", nil)
}

/**
删除说说
*/
func DeleteChat(ctx iris.Context) {
	chat := models.Chat{}
	fields := []string{"id"}
	validateFields := []string{"ID"}
	_, err := controllers.GetRightModel(ctx, &chat, fields, validateFields)
	if err != nil {
		response.RenderError(ctx, err.Error(), nil)
		return
	}
	res := models.DeleteChat(&chat)
	if !res {
		response.RenderError(ctx, "删除说说失败，请稍后再试！", nil)
		return
	}
	response.RenderSuccess(ctx, "删除说说成功！", nil)
}
